### DISCUSSION TITLE
bpftrace performance question

### DISCUSSION THREAD

**Discussion by Engg-blogger** [bpftrace/bpftrace#3098] [General]
Hi Team - After running some performance tests on bpftrace, it looks like bpftrace process consumes 150 MB of RES memory and 300 MB of VIRT memory in ubuntu 20. Is there a way to optimize the RES memory for smaller machines. I am just running tcp connect.

**Comment by danobi** [comment]
Hi, stay tuned for AoT bpftrace:

* https://github.com/bpftrace/bpftrace/issues/2918
* https://dxuuu.xyz/aot-bpftrace.html

It'll help reduce resource requirements as well as improve startup time

**Reply by Engg-blogger** [reply]
Sure, What is the expected completion time for this initiative? How much % memory consumption we expect this will reduce?

**Reply by danobi** [reply]
Arthur is looking at finishing up the prototype. That should be relatively quick (on the order of months). The long tail is supporting all of bpftrace's features. A lot of features will need to be examined and reworked to support AOT. We plan on prioritizing based on community feedback.

Not sure about exact numbers for memory savings but it should be significant. AOT binaries will not be running LLVM.

**Comment by Engg-blogger** [comment]
@danobi Whats the status on this feature. Will it be fully ready anytime soon?

**Reply by danobi** [reply]
Fully ready, probably not very soon. But something that functions for subset of features, yes probably. If you have specific features you need  please let me know.

**Reply by Engg-blogger** [reply]
@danobi For now, we are using tcpconnect scripts and tracing udp messages. But also want to trace system processes and all.
